### ISSUE TITLE
testing-library can't find i18next Trans texts

### ISSUE THREAD

**Issue by davydspotnana** [callstack/react-native-testing-library#964]
## Describe the bug

I'm using a [`<Trans/>` component](https://react.i18next.com/latest/trans-component) from `react-i18next` module to translate complex react elements (without mocking it) 

It works well in React Native runtime:
![image](https://user-images.githubusercontent.com/95620376/165528491-082d9af8-6dac-49f3-99aa-3d59f2d9a18a.png)

- But when I tried to find elements by text it doesn't work 
- When I used `debug()` text that I'm trying to find exists

## Expected behavior

```tsx
//...
const screen = render(<Text><Trans t={t}>My test</Trans></Text>)

screen.getByText('My test') // will work
```

## Steps to Reproduce

```
git clone https://github.com/retyui/testing-library-and-i18next-issue
cd testing-library-and-i18next-issue
yarn
yarn test
```

## Screenshots

Full code you find here https://github.com/retyui/testing-library-and-i18next-issue

```tsx
      <Text>
        <Trans t={t}>My test</Trans>
      </Text>

      <Text>
        <Trans t={t}>
          Open <Text style={styles.link}>link</Text>
        </Trans>
      </Text>

      <Text>
        <Trans t={t} values={{name: 'callstack'}}>
          Hello <Text style={styles.bold}>{'{{name}}'}</Text>!
        </Trans>
      </Text>
```

![image](https://user-images.githubusercontent.com/95620376/165529752-a914e31a-03b1-4992-af29-e6d825a85db1.png)

## Versions

```sh
  npmPackages:
    @testing-library/react-native: ^9.1.0 => 9.1.0 
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.0 => 0.68.0 
    react-test-renderer: 17.0.2 => 17.0.2 
```

**Comment by pierrezimmermannbam** [comment]
Hi @davydspotnana , this issue is currently being adressed (see #937 for more information). You can refer to this comment https://github.com/callstack/react-native-testing-library/issues/937#issuecomment-1078424856 for a workaround by mocking the trans component. Hope this helps !

**Comment by AugustinLF** [comment]
Thanks the help @pierrezimmermannbam, will mark as duplicate.

**Comment by retyui** [comment]
@pierrezimmermannbam I am an author of two these issue ) I don think that this is duplicate (@AugustinLF please reopen it)

I don't want to mock a `Trans` component because it affect testing 



With a mock I can't:
1) testing pluralization
```tsx
<Trans values={{cars: 2}}>{`{{cars}} car`}`</Trans> 
// getByText('2 cars') 
// getByText('1 car')
```

- Verify values
```tsx
<Trans values={{ NON__CAR_NAME: 2 }}>{`{{cars}} car`}`</Trans> 
// getByText('2 cars') // not found
```

**Comment by AugustinLF** [comment]
@retyui  The root cause is the same in both issues. RNTL can only detect text directly inside `<Text>` components. If you have any intermediate component (even if it doesn't appear in the renderer) RNTL will be confused and won't be able to match it.

We agree that mocking is not a long term solution, we're still trying to find a real fix, there's some progress in #967 but it's a non trivial issue.
